energy-integration/governance-v2/src/lib.rs

```diff
@@ -25,34 +27,76 @@ pub trait GovernanceV2:
     + views::ViewsModule
     + energy_query::EnergyQueryModule
 {
-    /// Used to deposit tokens for "payable" actions.
-    /// Funds will be returned if the proposal is defeated.
-    /// To keep the logic simple, all tokens have to be deposited at once
+    /// Used to deposit tokens to gather threshold min_fee.
+    /// Funds will be returned if the proposal is canceled.
     #[payable("*")]
     #[endpoint(depositTokensForProposal)]
     fn deposit_tokens_for_proposal(&self, proposal_id: ProposalId) {
         self.require_caller_not_self();
         self.require_valid_proposal_id(proposal_id);
 
-        let depositor_mapper = self.payments_depositor(proposal_id);
-        require!(depositor_mapper.is_empty(), "Payments already deposited");
-
-        let required_payments = self.required_payments_for_proposal(proposal_id).get();
         require!(
-            !required_payments.is_empty(),
-            "This proposal requires no payments"
+            !self.proposal_reached_min_fees(proposal_id),
+            MIN_FEES_REACHED
         );
 
-        let actual_payments = self.call_value().all_esdt_transfers();
+        let additional_fee = self.call_value().single_esdt();
         require!(
-            actual_payments == required_payments,
-            "Invalid payments, must match the required payments"
+            self.governance_token_id()
+                .get()
+                .eq(&additional_fee.token_identifier),
+            WRONG_TOKEN_ID
         );
 
         let caller = self.blockchain().get_caller();
-        depositor_mapper.set(&caller);
+        let mut proposal = self.proposals().get(proposal_id);
+        proposal.fees.entries.push(FeeEntry {
+            depositor_addr: caller.clone(),
+            tokens: additional_fee.clone(),
```

Comment:
Should probably also have a minimum amount per deposit, so the SC can't get stuck if someone keeps spamming payments of 0.000000....1 tokens.